### DISCUSSION TITLE
Problems with floating edges

### DISCUSSION THREAD

**Discussion by activey** [bcakmakoglu/vue-flow#966] [Q&A] [answered]
Hi :)

Currently I'm migrating from react-flow to vue-flow and stumbled upon a problem with floating edges support. I've basically copied the example from here https://codesandbox.io/s/vue-flow-floating-edges-zfxdok into my codebase with small adjustments, I'm using the same createElements() just to see how it works on my local machine and I'm observing the given behavior: edges are rendered correctly but there is a flood of error messages in the webbrowser console like this:

```
Error: <path> attribute d: Unexpected end of attribute. Expected number, "….809482216579454".
p
```
It's literally a flood. Another thing is that when I switch to my own dataset the edges are rendered chaotically like on the screenshot attached.

![image](https://github.com/bcakmakoglu/vue-flow/assets/1751891/baa9dbc5-cbdb-4584-a24b-553e86d98ca0)

I use an ES6 syntax, not TypeScript, so I had to modify the floating-edge-utils file to make the syntax valid.
I appreciate any help in this area. Thanks :)

**Comment by bcakmakoglu** [comment]
You're trying to use a heavily outdated example (no idea where you even found the sandbox still 🤣).
[Use the example *in the repo* instead.](https://github.com/bcakmakoglu/vue-flow/tree/master/examples/vite/src/FloatingEdges)

**Comment by activey** [comment]
Oh, right, I've aligned my code to the latest repo version and the errors are no longer there :)
But, my diagram edges looks abnormal, swinging in the air. It all looks fine in react version :/ 
I use nested nodes, maybe that's a problem? :(

**Reply by bcakmakoglu** [reply]
No, nested nodes are irrelevant to this.
You're probably doing something wrong as the example works completely fine.
Are you sure you're using the first element from the array returned by `getBezierPath`?

**Comment by activey** [comment]
Heh, I've noticed something super interesting :) My diagram consist of one parent node where the majority of the nodes are inside (only one is outside the parent), when you move nodes inside the parent the edges are rendered all fine, they switch places according to nodes position and everything looks all fine, but, when you just move parent node it all breaks. Please see the video attached.

https://github.com/bcakmakoglu/vue-flow/assets/1751891/a8d6845a-4939-4d5c-9e32-e5d7668ca92e

**Reply by bcakmakoglu** [reply]
I will need a sandbox reproduction for this. A Video can't really help me debug this issue for you.
[Here's a template.](https://codesandbox.io/p/sandbox/vue-flow-basic-gfgro4)

(Also use the reply box of each answer instead of writing a whole new answer, it just clutters this thread instead of creating one continuous thread ^^)

**Reply by activey** [reply]
Ah, sorry, my bad. Okay, I will prepare a sandbox for this case in the afternoon :)

**Reply by activey** [reply]
@bcakmakoglu here is the sandbox: https://codesandbox.io/p/sandbox/fancy-shape-4b4hd6

**Reply by bcakmakoglu** [reply, marked as answer]
Alright, so the problem is somewhat related to nested nodes.
With nested nodes, we have to rely on `computedPosition` instead of just `position`.
So you will have to update the utils to use *that* property instead of the `position` prop.

```ts
// this helper function returns the intersection point
// of the line between the center of the intersectionNode and the target node
function getNodeIntersection(intersectionNode: GraphNode, targetNode: GraphNode): XYPosition {
  // https://math.stackexchange.com/questions/1724792/an-algorithm-for-finding-the-intersection-point-between-a-center-of-vision-and-a

// use computedPosition instead of position here
  const {
    dimensions: { width: intersectionNodeWidth, height: intersectionNodeHeight },
    computedPosition: intersectionNodePosition,
  } = intersectionNode
  const targetPosition = targetNode.computedPosition

  const w = intersectionNodeWidth / 2
  const h = intersectionNodeHeight / 2

  const x2 = intersectionNodePosition.x + w
  const y2 = intersectionNodePosition.y + h
  const x1 = targetPosition.x + w
  const y1 = targetPosition.y + h

  const xx1 = (x1 - x2) / (2 * w) - (y1 - y2) / (2 * h)
  const yy1 = (x1 - x2) / (2 * w) + (y1 - y2) / (2 * h)
  const a = 1 / (Math.abs(xx1) + Math.abs(yy1))
  const xx3 = a * xx1
  const yy3 = a * yy1
  const x = w * (xx3 + yy3) + x2
  const y = h * (-xx3 + yy3) + y2

  return { x, y }
}

// returns the position (top,right,bottom or right) passed node compared to the intersection point
function getEdgePosition(node: GraphNode, intersectionPoint: XYPosition) {

// use computedPosition instead of position
  const n = { ...node.computedPosition, ...node.dimensions }
  const nx = Math.round(n.x)
  const ny = Math.round(n.y)
  const px = Math.round(intersectionPoint.x)
  const py = Math.round(intersectionPoint.y)

  if (px <= nx + 1) {
    return Position.Left
  }
  if (px >= nx + n.width - 1) {
    return Position.Right
  }
  if (py <= ny + 1) {
    return Position.Top
  }
  if (py >= n.y + n.height - 1) {
    return Position.Bottom
  }

  return Position.Top
}
```

[Here's a working sandbox.](https://codesandbox.io/p/sandbox/fancy-shape-4b4hd6)

**Reply by activey** [reply]
I think the link you've just sent leads to my broken sandbox ;)

**Reply by activey** [reply]
Nevertheless :slightly_smiling_face:  I've copied fragment of this code into my codebase and it worked :smile: Thank you very much for your help! Do you think the example code in vue-flow should be updated with this change?

**Reply by activey** [reply]
Btw ;> Instead of using raw `<g>` in FloatingEdge I switched to `<BaseEdge>` and it works quite well :}

**Reply by bcakmakoglu** [reply]
Yeah, I'll update the example as `computedPosition` would work for nested or non-nested nodes :)
